### PR TITLE
Ensure that tests run in both Karma and Node (p.s. they run really fast in node).

### PR DESCRIPTION
**Pull request by aljones15**
Remove crypto error & console.logs.

Generate a mock key.

Remove bedrock-web-kms from test project.

Switching back to get HMAC stuff.

Add id to mock HMAC.

Add should to eslintrc.js globals.

Closer to removing real kmsservice from project entirely.

Remove ExtendMockKeyService.

Fix await in wrapKey.

Remove need for bedrock-web-kms for karma.

Use kmsPlugin in mockMasterKey.

Remove code to stub out Pretender.

Mock tests (#2)

* Remove mocha setup.

* First test working in node.

* Ok a node test really succeeds this time.

* Move closer to using sinon for all mocks.

* All tests run in karma using sinon stubs.

* All tests passing in both node and karma.

* Added doc strings.

Addresses:
https://github.com/digitalbazaar/data-hub-client/issues/1

### CODE REVIEW

File: tests/test-karma.js [hunk 1]
```diff
@@ -4,4 +4,4 @@
  * Copyright (c) 2011-2019 Digital Bazaar, Inc. All rights reserved.
  */
 // load polyfills
-require('core-js/fn/promise');
+require('core-js/features/promise');
```
Comment (dlongley) [inline tests/test-karma.js]: Do we still need this?

File: tests/test-mocha.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// this is the polyfil so node has a crypto.subtle etc.
+const crypto = require('isomorphic-webcrypto');
+global.crypto = crypto;
```
Comment (dlongley) [inline tests/test-mocha.js]: We need to figure out the right webpack aliasing/whatever magic so we don't have to use `global`. Seems like we'd want some `crypto.js` file to pull in that, on node, exports `isomorphic-webcrypto` and in the browser exports `window.crypto || window.msCrypto` or something to that effect. Then we'd import `./crypto.js` in the various algorithm files.

cc: @davidlehn

Comment (aljones15) [reply tests/test-mocha.js]: oh yeah good idea webpack actually has a boolean that exposes crypto to the bundle. so I will look into that today.

Comment (aljones15) [reply tests/test-mocha.js]: @dlongley actually the mocha tests just use the source dir (not even babel is used). we could test on a webpack build though of the source with a polyfill though. webpack already provided a polyfil for node crypto for browsers the issue here is that these tests need subtle which is a browser only crypto library.

File: tests/test-karma.js (hunk 1 above)
Comment (aljones15) [reply tests/test-karma.js]: no we don't

File: tests/test-mocha.js (hunk 2 above)
Comment (dlongley) [reply tests/test-mocha.js]: Right -- we're doing this "browser-first", so we need to polyfill node, not the browser. There is a pattern where you can do aliasing with webpack such that my solution in comment https://github.com/digitalbazaar/data-hub-client/pull/3#discussion_r277301573 should work (I think). @davidlehn -- can you confirm?

Comment (aljones15) [reply tests/test-mocha.js]: I believe that webpack plugins https://webpack.js.org/guides/shimming/

would handle this. as webpack the existing setup:

```
node: {
  crypto: true
}
```

actually does the opposite i.e. it puts a node crypto polyfil in the browser.
https://webpack.js.org/configuration/node/

so yeah expand we need this:

```
plugins: [
     new webpack.ProvidePlugin({
       crypto: 'isomorphic-webcrypto'
     })
]
```

@davidlehn confirm?

Comment (aljones15) [reply tests/test-mocha.js]: @davidlehn basically make sure that the crypto stuff handled by webpack is being handled correctly here.

**Comment by dlongley** [comment]
Replaced by #5.

**Pull request closed without merging**